config/configuration.rb

```diff
@@ -0,0 +1,17 @@
+# dashboard app specific configuration
+class Configuration
+  class << self
+    attr_writer :app_development_enabled
```

Comment:
This suggestion is more optional...

Since the following method `Configuration.app_development_enabled?` is a conditional with a question mark and not a typical getter `Configuration.app_development_enabled`, maybe making the following methods: `Configuration.enable_app_development` and `Configuration.disable_app_development` might be more prudent than the setter `Configuration.app_development_enabled=`.

Reply:
This would allow us to force `@app_development_enabled` to always be a boolean.

Reply:
Will you clarify how this is "optimal" and "prudent"?

Reply:
Guarantees `Configuration.app_development_enabled?` always returns a boolean, e.g.:

```ruby
Configuration.app_development_enabled = "something"
Configuration.app_development_enabled? #=> "something"
```

also having a setter in my opinion makes me think I can call the getter...

```ruby
Configuration.app_development_enabled = true
Configuration.app_development_enabled #=> RAISES ERROR
```

especially if we add more configuration options in the future that have both setters and getters...

```ruby
Configuration.motd_path = "/etc/motd"
Configuration.motd_path #=> "/etc/motd"
```

Reply:
> Guarantees Configuration.app_development_enabled? always returns a boolean

In this case I don't think its a problem if it returns a boolean or an object since every object in Ruby has a truthy or falsy value.

I was asking for clarification because I thought your primary motivation for suggesting this change was:

> also having a setter in my opinion makes me think I can call the getter

This is just a preference and not a matter of "prudence" or "optimism".

Reply:
> not a matter of "prudence"

Hah, not sure debating the meaning of "prudent" is necessary here. I just wanted to point out that this may or may not lead to confusion in the future:

```ruby
Configuration.motd_path = "/etc/motd"
Configuration.motd_path #=> "/etc/motd"

Configuration.app_development_enabled = true
Configuration.app_development_enabled #=> RAISES ERROR
```

But as I initially said, it is just a suggestion. So you can just say it is not necessary for now and ignore it.

Reply:
Oops I misread your comment:

> This suggestion is more optional...

Thought you said:

> This suggestion is more optimal...